lib/nimble.phx.gen.template/template.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Nimble.Phx.Gen.Template.Template do
+  alias Nimble.Phx.Gen.Template.{Project, Addons}
+  alias Nimble.Phx.Gen.Template.Api.Template, as: ApiTemplate
+  alias Nimble.Phx.Gen.Template.Web.Template, as: WebTemplate
+
+  def apply(%Project{} = project) do
+    project = shared_setup(project)
+
+    mix_format()
+
+    if Project.api?(project) do
+      ApiTemplate.apply(project)
+    else
+      WebTemplate.apply(project)
+    end
+
+    mix_format()
+  end
+
+  # Common setup for both API and Web projects
+  defp shared_setup(%Project{} = project) do
+    project
+    |> Addons.Makefile.apply()
+    |> Addons.Docker.apply()
+    |> Addons.Github.apply()
+    |> Addons.TestingTool.apply()
+  end
+
+  defp mix_format do
+    Mix.shell().cmd("mix deps.get")
```

Comment:
Running `mix deps.get` in `mix_format` is really unexpected 😨  This should be in another method

Reply:
We don't need that anymore @olivierobert